catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ImportCommand.java

```diff
@@ -0,0 +1,212 @@
+
+    @Override
+    protected Object executeWithSubject() throws Exception {
+        ZipValidator zipValidator = initZipValidator();
+        File file = initImportFile(importFile);
+        InputTransformer transformer = getServiceByFilter(InputTransformer.class,
+                String.format("(%s=%s)",
+                        "id",
+                        DEFAULT_TRANSFORMER_ID)).orElseThrow(() -> new CatalogCommandRuntimeException(
+                "Could not get " + DEFAULT_TRANSFORMER_ID + " input transformer"));
+
+        if (!zipValidator.validateZipFile(importFile)) {
+            throw new CatalogCommandRuntimeException("Signature on zip file is not valid");
+        }
+
+        console.println("Importing file");
+        try (InputStream fis = new FileInputStream(file);
+                ZipInputStream zipInputStream = new ZipInputStream(fis)) {
+            ZipEntry entry = zipInputStream.getNextEntry();
+
+            while (entry != null) {
+                String filename = entry.getName();
+
+                if (filename.startsWith("META-INF")) {
+                    entry = zipInputStream.getNextEntry();
+                    continue;
+                }
+
+                String[] pathParts = filename.split("/");
+                String id = pathParts[ID];
+                String type = pathParts[TYPE];
+
+                switch (type) {
+                case "metacard": {
+                    String metacardName = pathParts[NAME];
+                    Metacard metacard = null;
+                    try {
+                        metacard = transformer.transform(new UncloseableBufferedInputStreamWrapper(
+                                zipInputStream), id);
+                    } catch (IOException | CatalogTransformerException e) {
+                        LOGGER.debug("Could not transform metacard: {}", id);
+                    }
+                    catalogProvider.create(new CreateRequestImpl(metacard));
+                    break;
+                }
+                case "content": {
+                    String contentFilename = pathParts[NAME];
+                    ContentItem contentItem = new ContentItemImpl(id,
+                            new ZipEntryByteSource(new UncloseableBufferedInputStreamWrapper(zipInputStream)),
+                            null,
+                            contentFilename,
+                            entry.getSize(),
+                            null);
+                    CreateStorageRequestImpl createStorageRequest = new CreateStorageRequestImpl(
+                            Collections.singletonList(contentItem),
+                            id,
+                            new HashMap<>());
+                    storageProvider.create(createStorageRequest);
+                    storageProvider.commit(createStorageRequest);
+                    break;
+                }
+                case "derived": {
+                    String qualifier = pathParts[NAME];
```

Comment:
`pathParts[NAME]` is used in all cases and could be moved out of the switch statement